src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java

```diff
@@ -4029,20 +4038,22 @@ private void processFileChanges(
         } else if (directory.startsWith(configPath)
                 && !isGeneratedConfigFile(fileChanged, configDirectory, serverDirectory)) { // config
                                                                                             // files
-            if (fileChanged.exists() && (changeType == ChangeType.MODIFY
-                    || changeType == ChangeType.CREATE)) {
+            if (fileChanged.exists() && (changeType == ChangeType.MODIFY || changeType == ChangeType.CREATE)) {
+                boolean generateFeaturesSuccess = false;
+                if ((fileChanged.getName().equals("server.xml")) && serverXmlFileParent == null && generateFeatures) {
+                    // server.xml modified
+                    generateFeaturesSuccess = incrementGenerateFeatures();
+                }
                 // suppress install feature warning - property must be set before calling copyConfigFolder
                 System.setProperty(SKIP_BETA_INSTALL_WARNING, Boolean.TRUE.toString());
-                copyConfigFolder(fileChanged, configDirectory, null);
+                // copyConfigFolder will install features if new ones are detected
+                copyConfigFolder(fileChanged, configDirectory, null, generateFeaturesSuccess);
```

Comment:
same as above, even if generate features fail we would still want to attempt to install any user specified features